src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -228,6 +249,53 @@ Task DisposeCoreInternalAsync()
         }
     }
 
+    internal async Task<bool> RebuildProjectAsync(
+        CancellationToken cancellationToken)
+    {
+        Assumes.NotNull(_project.Services.HostObject);
+        await _projectThreadingService.SwitchToUIThread();
+
+        if (_vsSolutionBuildManager2 is null)
+        {
+            _vsSolutionBuildManager2 = await _vsSolutionBuildManagerService.GetValueAsync(cancellationToken);
+        }
+
+       
+        // Step 1: Register sbm events
+        var _ = _vsSolutionBuildManager2.AdviseUpdateSolutionEvents(this, out uint cookie);
```

Comment:
Should probably check the HResult returned here:

```suggestion
        Verify.HResult(_vsSolutionBuildManager2.AdviseUpdateSolutionEvents(this, out uint cookie));
```